internal/provider/resources/resource_user_invite.go

```diff
@@ -0,0 +1,352 @@
+		})
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	// Extract the inviter from the inviter object
+	var inviter models.SubjectProfile
+	if !data.Inviter.IsUnknown() && !data.Inviter.IsNull() {
+		diags := data.Inviter.As(ctx, &inviter, basetypes.ObjectAsOptions{
+			UnhandledNullAsEmpty:    false,
+			UnhandledUnknownAsEmpty: false,
+		})
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	// Get the user invite
+	user, err := r.IamClient.GetUserWithResponse(
+		ctx,
+		r.OrganizationId,
+		invitee.Id.ValueString(),
+	)
+	if err != nil {
+		tflog.Error(ctx, "failed to get User Invite", map[string]interface{}{"error": err})
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to get User Invite, got error: %s", err),
+		)
+		return
+	}
+	statusCode, diagnostic := clients.NormalizeAPIError(ctx, user.HTTPResponse, user.Body)
+	// If the resource no longer exists, it is recommended to ignore the errors
+	// and call RemoveResource to remove the resource from the state. The next Terraform plan will recreate the resource.
+	if statusCode == http.StatusNotFound {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if diagnostic != nil {
+		resp.Diagnostics.Append(diagnostic)
+		return
+	}
+
+	// Generate userInvite from the get user API response
+	userInvite := iam.Invite{
+		ExpiresAt: data.ExpiresAt.ValueString(),
+		InviteId:  data.InviteId.ValueString(),
+		Invitee: iam.BasicSubjectProfile{
+			ApiTokenName: invitee.ApiTokenName.ValueStringPointer(),
+			AvatarUrl:    invitee.AvatarUrl.ValueStringPointer(),
+			FullName:     invitee.FullName.ValueStringPointer(),
+			Id:           invitee.Id.ValueString(),
+			SubjectType:  lo.ToPtr(iam.BasicSubjectProfileSubjectType(invitee.SubjectType.ValueString())),
+			Username:     invitee.Username.ValueStringPointer(),
+		},
+		Inviter: iam.BasicSubjectProfile{
+			ApiTokenName: inviter.ApiTokenName.ValueStringPointer(),
+			AvatarUrl:    inviter.AvatarUrl.ValueStringPointer(),
+			FullName:     inviter.FullName.ValueStringPointer(),
+			Id:           inviter.Id.ValueString(),
+			SubjectType:  lo.ToPtr(iam.BasicSubjectProfileSubjectType(inviter.SubjectType.ValueString())),
+			Username:     inviter.Username.ValueStringPointer(),
+		},
+		OrganizationId: r.OrganizationId,
+		UserId:         lo.ToPtr(user.JSON200.Id),
+	}
+
+	diags := data.ReadFromResponse(ctx, &userInvite, data.Email.ValueString(), data.Role.ValueString())
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("read a User Invite resource: %v", data.InviteId.ValueString()))
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *UserInviteResource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var data models.UserInvite
+
+	// Delete existing user invite
```

Comment:
comment seems outta place

Reply:
maybe we can add this to above the function definition of what we're gonna do something like
```
1. Delete existing user invite
2. Create a new user invite
<reason why>
```

Reply:
added the comment to make it clear that we delete the existing user invite first and then create a new user invite (since we don't have an update user invite endpoint)